src/components/ListingDetails/listingDetaild.types.d.ts

```diff
@@ -0,0 +1,12 @@
+import { IListingProps } from '../../sections/Listings/listings.types';
+
+/**
+ * Types for ListingDetails
+ * @param {id} string - Id of the listing
+ * @param {data} IListingProps[] - Listing data
+ */
+
+export interface IListingDetails {
+  id: string | undefined;
```

Comment:
why not make it optional?

Reply:
https://github.com/igorizviekov/mt-booking_app/pull/29/commits/d1c9eba7d4148f02c34d0999970b07bd034aa9cd

i thought it would better to make it required as without id prop react router will render another page anyway.